tools/go-agent/tools/vendor.go

```diff
@@ -0,0 +1,31 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package tools
+
+import "strings"
+
+const vendorDir = "/vendor/"
+
+// UnVendor removes the vendor directory from the path.
+func UnVendor(path string) string {
```

Comment:
Shall we cover this with tests? Also, will `/a/vendor` won't be trimmed?

Finally, would be good to explain how `/a/vendor/b` -> `/ab` is meaningful.

Reply:
> Shall we cover this with tests?

Sure, I will create a PR for this.

> Also, will `/a/vendor` won't be trimmed?

Yes, It won't be trimmed. Because `vendor` needs to be part of the path in order to take effect.

> Finally, would be good to explain how `/a/vendor/b` -> `/ab` is meaningful.

`/b` should be meaningful, because when we do an instrument, we only care about the actual package name, such as `github.com/apache/skywalking-go/agent/core`, but the vendor management project will add a `vendor` package prefix for this, so we need to trim it.